test-case/check-pause-resume-suspend-resume.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+##
+## Case Name: check-pause-resume-suspend-resume
+## Preconditions:
+##    N/A
+## Description:
+##    test playback with pause/resume as well as suspend/resume
+##    fake pause/resume with expect on aplay process
+##    when aplay is paused, have system enter suspend state for 5 secs
+##    resume from suspend state
+##    release aplay from paused state, then pause again
+##    repeat
+## Case step:
+##    1. aplay process is started
+##    2. aplay process is then paused via mock spacebar press via expect
+##    3. once aplay is paused, have system enter suspend state for 5 secs
+##    4. resume system from suspend state and unpause aplay process
+##    5. repeat from #2 - #4 50x
+## Expect result:
+##    no errors occur for either process
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['p']='pcm'          OPT_DESC_lst['p']='playback pcm. Example: hw:0,0'
+OPT_PARM_lst['p']=1             OPT_VALUE_lst['p']='hw:0,0'
+
+OPT_OPT_lst['l']='loop'         OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1             OPT_VALUE_lst['l']=5
+
+OPT_OPT_lst['i']='sleep-period' OPT_DESC_lst['i']='sleep period of aplay, unit is ms'
+OPT_PARM_lst['i']=1             OPT_VALUE_lst['i']='100'
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+repeat_count=${OPT_VALUE_lst['l']}
+#TODO: file name salt for capture
+# configure random value range
+sleep_period=${OPT_VALUE_lst['i']}
+
+cmd=aplay
+dummy_file=/dev/zero
+fmt="S16_LE"
+channel=2
+rate=48000
+pcm=${OPT_VALUE_lst['p']}
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+func_lib_setup_kernel_last_line
+
+dlogi "Entering playback expect script with: $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q"
+dlogi "Will enter suspend-resume cycle during paused period of aplay process"
+
+rm -rf /tmp/sof-test.lock
+
+# expect is tcl language script
+#   expr rand(): produces random numbers between 0 and 1
+#   after ms: Ms must be an integer giving a time in milliseconds.
+#       The command sleeps for ms milliseconds and then returns.
+expect <<PLAYBACK
+spawn $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q
+set i 1
+expect {
+    "#*+*\%" {
+        #aplay is playing now
+        set sleep_t $sleep_period
+        puts "\r(\$i/$repeat_count) pb_pbm: Pause aplay, then wait for"
+        puts "\r(\$i/$repeat_count) pb_pbm: $sleep_t ms after pause"
+        send " "
+        after \$sleep_t
+        puts "Finished sleep after pause; entering suspend-resume cycle"
+
+        #enter suspend-resume cycle once per pause instance
```

Comment:
can we add a check for the pipeline status here to make sure we entered **pause** state before we go into suspend.

Reply:
absolutely. will do.

Reply:
Done.